target/linux/ramips/dts/mt7628an_qding_qc202.dts

```diff
@@ -0,0 +1,116 @@
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "qding,qc202", "mediatek,mt7628an-soc";
+	model = "Qding QC202";
+
+	leds {
+		compatible = "gpio-leds";
+
+		gpio-relay {
+			label = "gpio-relay";
+			gpios = <&gpio 42 GPIO_ACTIVE_HIGH>;
+		};
+
+		ethernet {
+			label = "ethernet";
+			gpios = <&gpio 43 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi {
+			label = "wifi";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "reset";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "gpio";
+		function = "gpio";
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+
+				nvmem-layout {
+					compatible = "fixed-layout";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					eeprom_factory_0: eeprom@0 {

```

Comment:
I don't see where you assigned the ethernet mac address in dts. Could you check what mac address is used in the vendor firmware for ethernet and wifi and assign it in dts? The mac address for these mtk mips devices are usually at 0x4 or somewhere around 0x20. You can hexdump the factory content to find them.

Reply:
Yes, its 0x04

![image](https://github.com/user-attachments/assets/094cb57f-4ae5-4dbb-a618-c17a756a270a)